sormas-ui/src/main/java/de/symeda/sormas/ui/labmessage/LabMessageController.java

```diff
@@ -401,141 +412,181 @@ private void editPatogenTest(SampleDto sampleDto, PathogenTestDto testDto, int c
+		fillSample(sampleDto, labMessageDto);
+		Window window = VaadinUiUtil.createPopupWindow();
+		CommitDiscardWrapperComponent<SampleCreateForm> sampleCreateComponent = getSampleCreateComponent(sampleDto, labMessageDto, window);
+		showFormWithLabMessage(labMessageDto, sampleCreateComponent, window, I18nProperties.getString(Strings.headingCreateNewSample));
 	}
 
-	private void createSample(SampleDto sampleDto, LabMessageDto labMessageDto) {
+	private void fillSample(SampleDto sampleDto, LabMessageDto labMessageDto) {
 		sampleDto.setSampleDateTime(labMessageDto.getSampleDateTime());
 		if (labMessageDto.getSampleReceivedDate() != null) {
 			sampleDto.setReceived(true);
 			sampleDto.setReceivedDate(labMessageDto.getSampleReceivedDate());
 			sampleDto.setLabSampleID(labMessageDto.getLabSampleId());
 		}
 		sampleDto.setSampleMaterial(labMessageDto.getSampleMaterial());
-		sampleDto.setSpecimenCondition(labMessageDto.getSpecimenCondition());
+		sampleDto.setSpecimenCondition(SpecimenCondition.ADEQUATE);
+		sampleDto.setLab(getLabReference(labMessageDto));
+		sampleDto.setLabDetails(labMessageDto.getTestLabName());
+	}
 
+	private FacilityReferenceDto getLabReference(LabMessageDto labMessageDto) {
+		FacilityFacade facilityFacade = FacadeProvider.getFacilityFacade();
 		List<FacilityReferenceDto> labs =
-			FacadeProvider.getFacilityFacade().getByExternalIdAndType(labMessageDto.getTestLabExternalId(), FacilityType.LABORATORY, false);
+				facilityFacade.getByExternalIdAndType(labMessageDto.getTestLabExternalId(), FacilityType.LABORATORY, false);
 		if (labs != null && labs.size() == 1) {
-			sampleDto.setLab(labs.get(0));
+			return labs.get(0);
+		} else {
+			return facilityFacade.getFacilityReferenceByUuid(FacilityDto.OTHER_FACILITY_UUID);
 		}
+	}
 
-		Window window = VaadinUiUtil.createPopupWindow();
+	private CommitDiscardWrapperComponent<SampleCreateForm> getSampleCreateComponent(SampleDto sampleDto, LabMessageDto labMessageDto, Window window) {
 		CommitDiscardWrapperComponent<SampleCreateForm> sampleCreateComponent =
-			ControllerProvider.getSampleController().getSampleCreateComponent(sampleDto, () -> {
-			});
+			ControllerProvider.getSampleController().getSampleCreateComponent(sampleDto, () -> { });
 
 		CheckBox checkBox = sampleCreateComponent.getWrappedComponent().getField(Captions.sampleIncludeTestOnCreation);
 		checkBox.setValue(Boolean.TRUE);
 		checkBox.setEnabled(false);
 		((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_RESULT)).setValue(labMessageDto.getTestResult());
 		((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TEST_TYPE)).setValue(labMessageDto.getTestType());
-		((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TESTED_DISEASE)).setValue(labMessageDto.getTestedDisease());
+		((ComboBox) sampleCreateComponent.getWrappedComponent().getField(PathogenTestDto.TESTED_DISEASE)).setValue(Disease.CORONAVIRUS);
```

Comment:
This is generic for the DEMIS adapter, not for labMessages in general. Please move setting such default values to the adapter.